lib/phoenix_live_view/utils.ex

```diff
@@ -98,13 +98,13 @@ defmodule Phoenix.LiveView.Utils do
   defp prune_uri(:not_mounted_at_router), do: :not_mounted_at_router
 
   defp prune_uri(url) do
-    %URI{host: host, port: port, scheme: scheme} = url
+    uri = URI.parse(url)
 
-    if host == nil do
+    if uri.host == nil do
       raise "client did not send full URL, missing host in #{url}"
     end
 
-    %URI{host: host, port: port, scheme: scheme}
+    uri
```

Comment:
```suggestion
    %{uri | path: nil}
```

Just to be sure we ignore any path.